Switch to using main/release rather than develop/main (#133)
pantheon-systems/pantheon-hud: CONTRIBUTING.md

## Release Process

> 1. Make a release commit:
> 1. From `main`, checkout a new branch `release_X.Y.Z`.

Reviewer: ```suggestion
1. From `main`, checkout a new branch `release_X.Y.Z`.
1. Make a release commit:
```
This was correct.

    * Drop the `-dev` from the version number in `package.json`, `README.md`, `readme.txt`, and `pantheon-hud.php`.

>     * Update the "Latest" heading in the changelog to the new version number with the date

Reviewer: Other plugins, we've switched to doing a X.Y.Z-dev heading that is search-replaced instead of Latest.

    * Commit these changes with the message `Release X.Y.Z`
    * Push the release branch up.
1. Open a Pull Request to merge `release_X.Y.Z` into `release`. Your PR should consist of all commits to `main` since the last release, and one commit to update the version number. The PR name should also be `Release X.Y.Z`.

> 1. After all tests pass and you have received approval from a CODEOWNER (including resolving any merge conflicts), merge the PR into `release`.

Reviewer: ```suggestion
1. After all tests pass and you have received approval from a CODEOWNER (including resolving any merge conflicts), merge the PR into `release`. Use a "merge" commit, do no not rebase or squash.
```

1. Wait for the [_Release pantheon-hud plugin to wp.org_ action](https://github.com/pantheon-systems/pantheon-hud/actions/workflows/wordpress-plugin-deploy.yml) to finish deploying to the WordPress.org plugin repository. If all goes well, users with SVN commit access for that plugin will receive an emailed diff of changes.
1. Check WordPress.org: Ensure that the changes are live on [the plugin repository](https://wordpress.org/plugins/pantheon-hud/). This may take a few minutes.
1. Following the release, prepare the next dev version with the following steps:
    * `git checkout develop`
    * `git rebase release`
    * Update the version number in all locations, incrementing the version by one patch version, and add the `-dev` flag (e.g. after releasing `1.2.3`, the new verison will be `1.2.4-dev`)

>     * Add a new `** Latest **` heading to the changelog

Reviewer: ```suggestion
    * Add a new `** X.Y.Z-dev **` heading to the changelog where `X.Y.Z-dev` is the new patch version.
```
